File: Python/lark_card_kit/models/elements/image.py
"""Image element for displaying images."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from .base import Element


@dataclass
class Image(Element):
    """Image element for displaying images.

    Displays an image from a URL.
    """

    tag: str = field(default="img", init=False)
    src: str = ""
    alt: str | None = None
    preview_src: str | None = None
    size: str | None = None
    mode: str | None = None
    width: str | None = None
    height: str | None = None
    corner_radius: str | None = None
    hover_tips: Any = None
    behaviors: list[Any] | None = None

File: TypeScript/LarkCardKit/src/models/elements/Image.ts
import type { ElementBase } from './Element';
import type { PlainText } from './PlainText';

export interface Image extends ElementBase {
  tag: 'img';
  img_key: string;
  alt?: PlainText;
  size?: string;
  mode?: string;
  width?: string;
  height?: string;
  preview?: boolean;
  hover_tips?: PlainText;
  click_url?: string;
}
